Feature/us4-recipe-details

ivanmansilla opened the pull request:
I have created two new components. Now we have the possibility for users to add reviews of the recipes. At first it was programmed to be in a separate column where they appear directly, but finally they appear in an Offcanvas

ivanmansilla committed:
feat: create new component comment

ivanmansilla committed:
feat: Comment component in progress, styles and responsive

ivanmansilla committed:
feat: Comment list functional

ivanmansilla committed:
feat: UseEffect, and use mocks to show comments list

ivanmansilla committed:
feat: Add animation, control errors and better design

ivanmansilla committed:
fix: Add padding to ingredients and steps box

ivanmansilla committed:
feat: Create component and post comment button

ivanmansilla committed:
feat: PostComment component done as a form modal

ivanmansilla committed:
feat: Implemtnation of Offcanvas in progress

ivanmansilla committed:
feat: header of comments transform into Offcanvas

ivanmansilla committed:
feat: List of reviews without border and better design

ivanmansilla committed:
feat: Post Comment button added in Offcanvas and modify title

ivanmansilla committed:
Change spansih for english in titles

ivanmansilla committed:
feat: Merge branch of comments

ivanmansilla commented:
The first version, it was comments, and it looked like this:
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/daec2764-5fe5-4469-b25b-b328df18417a)

ivanmansilla commented:
At this moment, I have modified and now it is an Offcanvas, I have aligned with develop and I have set it to appear on the right when click the button - Toggle Comment:
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/8639e10f-61ac-41e9-9871-33145f0c10ae)

ivanmansilla committed:
fix: Aligned with develop and Offcanvas to the right

ivanmansilla commented:
To do so far, what I have done is to create a new component called Comments, which is the reviews of the people, thanks to a map where it is showing them. Right now a mock is used but it is prepared for when we make the fetch bringing the reviews it shows them. 
If we click on the post comment button (it will be review) it shows us a modal, which is another new component that I have created called PostComment.
This is its visualization. The username, will come by props and we will not ask it to the user, and the rating will be added.
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/97b490aa-3bea-4f4b-88e6-ff7caee42dd2)

ivanmansilla committed:
Merge branch 'develop' of https://github.com/UB-ES-2023-A2/kasula-frontend into feature/comments-recipe-detail

ivanmansilla committed:
feat: Reviews and username through fetch

BlessedLongsword commented:
I don't know if @xvxnoah already told you this. But in the post comment modal I would make the following changes:

Post a comment -> Make a Review
Username -> Rating (instead of a written field, it should use a star system).

Does the file selection appear in spanish because it uses browser/system language?

ivanmansilla committed:
feat: rating and username in post review

ivanmansilla committed:
feat: in progress post review to bbdd

ivanmansilla committed:
feat: .env updated

ivanmansilla committed:
Merge branch 'develop' of https://github.com/UB-ES-2023-A2/kasula-frontend into feature/comments-recipe-detail

ivanmansilla committed:
feat: post done in reviews, and userfeed temporaly change validation

xvxnoah committed:
Update PostReview.jsx

BlessedLongsword committed:
Slight column change

BlessedLongsword committed:
Merge remote-tracking branch 'origin' into 161-us15-fe1-crete-collections-screen

Also fixed recipes image in recipe detail

ivanmansilla committed:
feat: post and get reviews

ivanmansilla committed:
feat: include rating in reviews and now is responsive

ivanmansilla committed:
Merge branch 'feature/comments-recipe-detail' into feature/us4-recipe-details

ivanmansilla commented:
Several changes have been made. In short, we already use the post and get for the reviews. Users can now post a review only, with an image or an optional comment, and also rate it from 1 to 5 stars. Also the get brings the information and is responsive. Then we will add a button to like the review for example.
It has also been updated as @BlessedLongsword commented, now it is review and in the post review the username field does not appear.
Comment that the first comments did not appear in the review because they were tests and we did not send them properly.
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/68d7c36e-ac5c-408d-8859-c281d9e58bbd)

ivanmansilla committed:
feat: likes in reviews in progress

BlessedLongsword committed:
Created collections screen and fixed navbar user dropdown

BlessedLongsword committed:
Merge branch 'feature/us4-recipe-details' into 161-us15-fe1-crete-collections-screen

BlessedLongsword committed:
Collections advancements:

Finished Collection page
Created CollectionView (needs testing)
Added colorless span style

BlessedLongsword committed:
Margin adjustments

BlessedLongsword committed:
Use deployed backend

ivanmansilla committed:
feat: New component Likes

ivanmansilla committed:
feat: funcionality Likes and control errors done

ivanmansilla committed:
feat: Callbacks to control re-renders

ivanmansilla committed:
Clean code

BlessedLongsword committed:
File Creation and Collection Create progress

ivanmansilla commented:
I have implemented the likes and all its functionality. It has cost me a lot, because it has much play with endpoints and various things, but finally I managed to add the functionality of likes, I tried to control all possible errors, by default will be 0 if you do not have likes. A user can only like a review once, and you can't like your own review. Attached is an image of how it looks like and another one of the error that pops up when you try to like something you have already liked.
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/0d8ccb43-99d9-4466-99ad-b773fbfcbee0)
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/2910c6fa-c984-43c7-95ad-4c69f85003ea)

xvxnoah commented:
> I have implemented the likes and all its functionality. It has cost me a lot, because it has much play with endpoints and various things, but finally I managed to add the functionality of likes, I tried to control all possible errors, by default will be 0 if you do not have likes. A user can only like a review once, and you can't like your own review. Attached is an image of how it looks like and another one of the error that pops up when you try to like something you have already liked. ![image](https://user-images.githubusercontent.com/72189801/284911440-0d8ccb43-99d9-4466-99ad-b773fbfcbee0.png) ![image](https://user-images.githubusercontent.com/72189801/284911507-2910c6fa-c984-43c7-95ad-4c69f85003ea.png)

Nice work Ivan! I have a couple of comments for you:

1. Can we add some margin between the number of likes and the icon?
2. Can we change the position of the like icon with the number? So we have the icon on the left and the number on its right side. That way we mirror the behaviour of the rating with the stars.
3. Maybe instead of showing that kind of error when trying to like again the review, we can just unlike the review (I will add that endpoint now to the backend as a hotfix).
4. Can we show the errors in a different way? It is just that I don't like to show the errors like web notifications.

xvxnoah commented:
> > I have implemented the likes and all its functionality. It has cost me a lot, because it has much play with endpoints and various things, but finally I managed to add the functionality of likes, I tried to control all possible errors, by default will be 0 if you do not have likes. A user can only like a review once, and you can't like your own review. Attached is an image of how it looks like and another one of the error that pops up when you try to like something you have already liked. ![image](https://user-images.githubusercontent.com/72189801/284911440-0d8ccb43-99d9-4466-99ad-b773fbfcbee0.png) ![image](https://user-images.githubusercontent.com/72189801/284911507-2910c6fa-c984-43c7-95ad-4c69f85003ea.png)
> 
> Nice work Ivan! I have a couple of comments for you:
> 
> 1. Can we add some margin between the number of likes and the icon?
> 2. Can we change the position of the like icon with the number? So we have the icon on the left and the number on its right side. That way we mirror the behaviour of the rating with the stars.
> 3. Maybe instead of showing that kind of error when trying to like again the review, we can just unlike the review (I will add that endpoint now to the backend as a hotfix).
> 4. Can we show the errors in a different way? It is just that I don't like to show the errors like web notifications.

By the way, I think the whole team would love a button to go back to the previous page in recipe details :))

ivanmansilla commented:
Perfect, I'll start with the comments you have told me! Regarding the issue of dislike I'm waiting and with the theme that comes out something other than an alert, I'll try to investigate something else, to see if I find something cool

ivanmansilla committed:
feat: likes icon on the left, and numbers like aligned witgned with stars

ivanmansilla committed:
feat: arrow left to back to the userfeed

ivanmansilla committed:
fix: nnavigate to userfeed

BlessedLongsword committed:
Collection creation, edition and deletion

- Changed component extensions

BlessedLongsword committed:
Merge branch 'develop' into 161-us15-fe1-crete-collections-screen

BlessedLongsword committed:
Add recipe to collection button

BlessedLongsword committed:
Preparing add to favorite

BlessedLongsword committed:
Adding recipes to favorite

BlessedLongsword committed:
Remove Recipes from collection

ivanmansilla committed:
feat: refactor design arrow and icon likes is heart

ivanmansilla committed:
feat: Dislike implementation done

ivanmansilla committed:
Merge branch 'develop' of https://github.com/UB-ES-2023-A2/kasula-frontend into feature/us4-recipe-details

ivanmansilla commented:
I have solved all the comments you made @noah. I have left a margin between icon and number of likes, which is now inverted. It is aligned next to the stars to make it look good. I have also changed the icon to heart, since the dislike is implemented. So when someone likes you, you can click again to remove your like without problems. Also apart from the dislikes I have implemented the arrow which takes you to the userfeed.
I attach images:
Like:
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/63b427d5-0fd6-4b9f-9cb4-93f069860a42)
Dislike:
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/e6972aa4-b273-4044-9408-158135702820)
Arrow:
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/460f0500-b3ac-473c-b2d3-e7cf0e67fbce)

xvxnoah committed:
Merge branch 'develop' into feature/us4-recipe-details

xvxnoah commented:
> I have solved all the comments you made @noah. I have left a margin between icon and number of likes, which is now inverted. It is aligned next to the stars to make it look good. I have also changed the icon to heart, since the dislike is implemented. So when someone likes you, you can click again to remove your like without problems. Also apart from the dislikes I have implemented the arrow which takes you to the userfeed. I attach images: Like: ![image](https://user-images.githubusercontent.com/72189801/285198900-63b427d5-0fd6-4b9f-9cb4-93f069860a42.png) Dislike: ![image](https://user-images.githubusercontent.com/72189801/285199361-e6972aa4-b273-4044-9408-158135702820.png) Arrow: ![image](https://user-images.githubusercontent.com/72189801/285199656-460f0500-b3ac-473c-b2d3-e7cf0e67fbce.png)

Good job Ivan! I will list here all the "bugs" and corrections that I think will be necessary for this sprint, we can also discuss them tonight after the daily:

1. At the moment a user that is not logged in can like and "unlike" a review.
2. If a user is not logged in, the "Post review" button should not be visible.
3. You should add the average rating of the recipe to the details.
4. The modal for posting a recipe looks good, but it would be nice that followed the same color pattern than the Post Recipe modal in the user feed.
5. As we commented before, you shouldn't be able to like your own review.
6. Users should be able to click on the review's image to see it bigger.
7. You should control how the review text is displayed because as you can see in this screenshot, the username text is on top the review:
![Screenshot 2023-11-23 at 18 32 26](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/47271218/b8a58905-7267-46bf-8978-34402714bc45)
8. For the button of Post review I would maybe follow the same colour as the button for posting a recipe.
9. We should consider using a different logo for the stars we want to give in the review, because now is the same logo as the difficulty in the recipe details, which might lead to confusions.

BlessedLongsword committed:
Add Recipe to collection

BlessedLongsword committed:
Merge pull request #188 from UB-ES-2023-A2/161-us15-fe1-crete-collections-screen

161-us15-fe1-crete-collections-screen

guzmanalejandro commented on package.json:
```diff
@@ -3,6 +3,8 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@fortawesome/free-solid-svg-icons": "^6.4.2",
```
We can use more icons of Fortawesome in the next sprint

guzmanalejandro commented on src/components/LikesReview.jsx:
```diff
@@ -0,0 +1,91 @@
+      }
+    }else{
+      
+      console.log(">>>By2: ", hasLikedByUser)
```
Are these logs necessary after merging the pull request?

ivanmansilla replied:
Yes,  I have forgotten to remove it

guzmanalejandro commented on src/components/LikesReview.jsx:
```diff
@@ -0,0 +1,91 @@
+    <Row>
+      <Col sm={3}>
+        <FontAwesomeIcon
+          icon={faHeart}
```
In the next sprint we could modify the like to change the empty heart icon to a filled one by liking it, for example.

guzmanalejandro commented on src/components/PostReview.jsx:
```diff
@@ -0,0 +1,136 @@
+    setImage(selectedImage);
+  };
+
+  const handlePostReview = async () => {
```
In this method we should control if the review is empty (the text) or not

guzmanalejandro replied:
Or control it from the backend and launch a popup, but I think it's more feasible to control it in the frontend

ivanmansilla replied:
The text is optional, can be empty

BlessedLongsword committed:
Merge branch 'feature/us4-recipe-details' into feature/us15-collections

BlessedLongsword committed:
Merge fallout

BlessedLongsword committed:
Merge pull request #189 from UB-ES-2023-A2/feature/us15-collections

Feature/us15 collections

ivanmansilla committed:
feat: Button post review only appear and can like/dislike if is logged

ivanmansilla committed:
feat: New average rating in more information and new icon to average

ivanmansilla committed:
feat: design of reviews and average recipe

ivanmansilla committed:
feat: Control errors with not possible review and click image of reviews

ivanmansilla commented:
I fixed all things that PO sayme, here the proves and recipeDetail finally:
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/de2dcd41-23a1-441f-8a1a-12c5e6a3c3d1)
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/ed1e9ab7-a598-4920-91ce-2ef61ad73ff2)
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/1a581150-b928-473e-859a-6f9f30b55864)
not logged:
![image](https://github.com/UB-ES-2023-A2/kasula-frontend/assets/72189801/1f6c66df-dee2-4891-8013-2dc5f2a43525)

ivanmansilla committed:
feat: reload recipeDetail with average

ivanmansilla committed:
fix: only two decimals

ivanmansilla committed:
fix: fixed 1 decimal

ivanmansilla committed:
fix: Control recipe is not null

xvxnoah reviewed (approved):
Looks awesome :))))
